src/main/java/org/elasticsearch/search/aggregations/bucket/significant/heuristics/ScriptHeuristic.java

```diff
@@ -62,32 +79,29 @@ public String getName() {
         }
     };
 
-    public ScriptHeuristic(ExecutableScript searchScript, String scriptLang, String scriptString, ScriptService.ScriptType scriptType, Map<String, Object> params) {
+    public ScriptHeuristic(ExecutableScript searchScript, Script script) {
```

Comment:
Feels a little ugly that `script` and `searchScript` are two different ways of specifying the code to run (one is compiled one is not) but there are no safeguards to ensure these are the same. Presumably the logic is that a constructor accepts one or the other arg but not both?